### ISSUE TITLE
Allow import/export from JSON

### ISSUE THREAD

**Issue by tversteeg** [cemsbv/pygef#136]
The current format is not serializable, so it can't be non-trivially sent over a network connection. We need to add `Cpt.from_json()` and `Cpt.to_json()` methods (same for Borehole).

**Comment by ritchie46** [comment]
Format of what? The gef files, or the state we keep internally when we parsed it?

**Comment by tversteeg** [comment]
> Format of what? The gef files, or the state we keep internally when we parsed it?

The whole `Cpt` and `Borehole` classes.

**Comment by ritchie46** [comment]
Right, there is room for optimization here. I think we should consider a naive `to_json`, but also a `serialize` / `deserialize`. (with some options, e.g. `parquet, ipc, compression types).

For instance the columnar format of the `DataFrame` can be stored in binary and be compressed. Which can save quite some wire transfer.

**Comment by breinbaas** [comment]
Why not use pydantic which makes it easy with json() and parse_raw

**Comment by oaydogmus** [comment]
Hello, is there any ongoing development regarding this issue?

**Comment by maarten-betman** [comment]
Hi @tversteeg, this feature would be useful for some services I'm running. Ok if I put a PR in for this enhancement?
How do you feel about usage of Pydantic?

**Comment by tversteeg** [comment]
Any PRs are greatly appreciated! Pydantic looks like a nice solution for this problem.
